source/particle/world.cc

```diff
@@ -516,30 +516,77 @@ namespace aspect
 
       // Load all particles from the data stream and store them in the local
       // particle map.
-      for (unsigned int i = 0; i < *n_particles_in_cell_ptr; ++i)
+      if (status == parallel::distributed::Triangulation<dim>::CELL_PERSIST)
         {
-          Particle<dim> p(pdata,property_manager->get_particle_size());
+          typename std::multimap<types::LevelInd,Particle<dim> >::const_iterator position_hint = particles.end();
+          for (unsigned int i = 0; i < *n_particles_in_cell_ptr; ++i)
+            {
+#ifdef DEAL_II_WITH_CXX11
+              position_hint = particles.emplace_hint(position_hint,
+                                                     std::make_pair(cell->level(),cell->index()),
+                                                     Particle<dim>(pdata,property_manager->get_particle_size()));
+#else
+              position_hint = particles.insert(position_hint,
+                                               std::make_pair(std::make_pair(cell->level(),cell->index()),
+                                                              Particle<dim>(pdata,property_manager->get_particle_size())));
+#endif
+              ++position_hint;
```

Comment:
so you just move the hint forward by one here, in anticipation of the next particle being inserted just after this one. that's correct of the next particle resides on the same cell. is there anything we could do better if we find out that the next particle is not on the same cell? (Just asking out of curiosity.)